native_client/ctcdecode/scorer.cpp

```diff
@@ -88,44 +88,54 @@ void Scorer::load_lm(const std::string& lm_path, const std::string& trie_path)
   const char* filename = lm_path.c_str();
   VALID_CHECK_EQ(access(filename, R_OK), 0, "Invalid language model path");
 
-  bool has_trie = trie_path.size() && access(trie_path.c_str(), R_OK) == 0;
-  // VALID_CHECK(has_trie, "Invalid trie path");
-
   lm::ngram::Config config;
   config.load_method = util::LoadMethod::LAZY;
   language_model_.reset(lm::ngram::LoadVirtual(filename, config));
 
+  uint64_t package_size;
+  {
+    util::scoped_fd fd(util::OpenReadOrThrow(filename));
+    package_size = util::SizeFile(fd.get());
+  }
+  uint64_t trie_offset = language_model_->GetEndOfSearchOffset();
+  bool has_trie = package_size > trie_offset;
```

Comment:
Does that means we support having no trie put into the package ? I think we removed that feature one or two commit ago ?

Reply:
It's just a check to be able to cleanly return with an error here instead of exploding in the `fin.read` when reading the trie. In the error handling cleanup commit I change this to check the flag and return an error code.